src/ChangeStream.php

```diff
@@ -101,6 +101,15 @@ public function next()
                 $this->hasAdvanced = true;
                 $this->resumeToken = $this->extractResumeToken($this->csIt->current());
             }
+            /* If the cursorId is 0, the server has invalidated the cursor so we
+             * will never perform another getMore. This means that we cannot
+             * resume and we can therefore unset the resumeCallable, which will
+             * free any reference to Watch. This will also free the only
+             * reference to an implicit session, since any such reference
+             * belongs to Watch. */
+            if ($this->getCursorId() === 0) {
```

Comment:
`getCursorId()` returns a `MongoDB\Driver\CursorId`, so this will never evaluate to true. You may need to cast the CursorId to a string and compare with `'0'`.